src/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```

Comment:
If we are ignoring a rule that means there is something wrong with this file :)

Reply:
Well, this is not always true. The rules sometimes are just recommendations or ways to check that you didn't forget something. :)

If you remove the comment and check it is just telling me that I'm early returning before some hooks... If I didn't make some mistake... Wich, I didn't. It is propositional.